server/middleware/users.coffee

```diff
@@ -74,9 +77,42 @@ module.exports =
     else
       throw new errors.Forbidden('Only admins and teachers can search by email')
 
-  removeFromClassrooms: wrap (req, res, next) ->
-    yield req.user.removeFromClassrooms()
-    next()
+  delete: wrap (req, res, userID) ->
+    userToDelete = yield database.getDocFromHandle(req, User)
+    if not userToDelete
+      throw new errors.NotFound('User not found.')
+    unless req.user?.isAdmin() or req.user?._id.equals(userToDelete._id)
+      throw new errors.Forbidden("Can't delete this user.")
+
+    yield userToDelete.removeFromClassrooms()
+    
+    # Delete personal subscription
+    if userToDelete.get('stripe.subscriptionID')
+      yield middleware.subscriptions.unsubscribeUser(req, userToDelete, false)
+    
+    # Delete recipient subscription
+    sponsorID = userToDelete.get('stripe.sponsorID')
+    if sponsorID
+      sponsor = yield User.findById(sponsorID)
+      if not sponsor
+        throw errors.UnprocessableEntity("Couldn't find subscription sponsor #{userToDelete.get('stripe.sponsorID')} of user #{req.userToDelete._id} to delete")
```

Comment:
`req.userToDelete._id` -> `userToDelete._id`. Should informally test this line (for example, run the test on this endpoint having deleted the sponsor, see if it returns 422).